lib/cell.rb

```diff
@@ -0,0 +1,49 @@
+class Cell
+
+  attr_reader :coordinate, :empty, :ship, :fired_upon
+
+  def initialize(coordinate)
+    @coordinate = coordinate
+    @empty = true
+    @ship = nil
+    @fired_upon = false
+  end
+
+  def empty?
+    @empty
+  end
+
+  def place_ship(ship)
+    @empty = false
+    @ship = ship
+  end
+
+  def fired_upon?
+    @fired_upon
+  end
+
+  def fire_upon
+    @fired_upon = true
+    if @ship == ship
+      ship.hit
+    end
```

Comment:
I feel like there must be a better way to do this.

Reply:
Maybe there is a way to not use an if statement. If its working then I say this looks great but maybe when we finish this project we can circle back to this for a different solution?

Reply:
I agree, if it works, let's keep going!